src/java.base/windows/classes/sun/nio/fs/WindowsFileStore.java

```diff
@@ -232,7 +232,15 @@ public boolean equals(Object ob) {
         if (!(ob instanceof WindowsFileStore))
             return false;
         WindowsFileStore other = (WindowsFileStore)ob;
-        return root.equals(other.root);
+        if (root.equals(other.root))
+            return true;
+        if (root.equalsIgnoreCase(other.root)) {
+             VolumeInformation otherInfo = other.volumeInformation();
+            return volInfo.volumeName().equals(otherInfo.volumeName()) &&
+                volInfo.fileSystemName().equals(otherInfo.fileSystemName()) &&
+                volInfo.volumeSerialNumber() == otherInfo.volumeSerialNumber();
+        }
+        return false;
```

Comment:
The volType == DRIVE_REMOTE case is the reason why we didn't fix this one in the past. Can you summarise that the VolumeInfomation is for this case?

Reply:
I was not able to test that case yet. Using the VolumeInformation was intended to limit spurious equality.

Can you explain abut DRIVE_REMOTE?

Reply:
> I was not able to test that case yet. Using the VolumeInformation was intended to limit spurious equality.
> 
> Can you explain abut DRIVE_REMOTE?

I think the volume type will be DRIVE_REMOTE when the root is a UNC (the constants for the volume types are defined in WIndowsConstants). I'm not 100% sure if the server and share names should be compared with or without regard to case so I think we have a bit more exploration to do there. The DRIVE_FIXED case is clear, and it would be good to at least get that fixed as there has been C:\ vs. c:\ confusion in a number of bug reports.

Reply:
Based on testing in a Win 10 VM, the volume type is in fact DRIVE_REMOTE. The volume serial number however is zero. From what I can test via testing and reading the server and share names don't care about case.

Reply:
If we are unsure, for UNC, perhaps the server and share name comparisons can remain case sensitive but the file path component comparison be case insensitive?

Reply:
Maybe this is something that the Microsoft folks can help us with, it might be that it depends on the transport.

Reply:
Good idea. I will ask them.